tutorials/nservicebus-101/lesson-2/tutorial.md

```diff
@@ -0,0 +1,191 @@
+
+A **message** is a collection of data sent via one-way communication between two endpoints. In NServiceBus, we can define a message via a simple class.
+
+In this lesson, we'll focus on the simpler type of message: commands. In [Lesson 4: Publishing events](../lesson-4/) we'll expand to look at events as well.
+
+Defining a command is pretty easy. We just create a class and mark it with the `ICommand` marker interface.
+
+snippet:Command
+
+The marker interface has no implementation, but lets NServiceBus know that the class is a command so that it can build up some metadata about the message type when the endpoint starts up. Any properties you create within the message constitute the message data.
+
+The name of the command class is also important. A command is an order to do something, so it should be named in the [imperative tense](https://en.wikipedia.org/wiki/Imperative_mood). `PlaceOrder`, `UpdateCustomerStatus`, and `ChargeCreditCard` are all great names for commands, because they are phrased as a command and are very specific. You can easily guess exactly what each of those messages will do. `UserMessage`, on the other hand, is not a good example. It is not in the imperative, and it's not very specific either. Another developer should know exactly what a command's purpose is just by reading the name.
+
+When sending a message, the endpoint's message serializer will serialize an instance of the `DoSomething` class (let's say to JSON) and add that to the contents of the outgoing message that goes to the queue. On the other end, the receiving endpoint will deserialize the message from JSON back to an instance of the message class so that it can be used in code.
+
+Messages can support whatever data the serializer can successfully process. Thus messages can even contain child objects or collections.
+
+snippet:ComplexCommand
+
+While it can be pretty tempting to throw in the kitchen sink, consider that messages are a contract between two endpoints. Any change to the message will likely involve a change on both the sender and receiver side. The more properties you have on a message, the more reasons it has to change, so keep your messages as slim as possible.
+
+Also, you should not embed logic within your NServiceBus message classes. Each message should contain only automatic properties and not computed properties or methods. Also, it is a good practice to instantiate collection properties from a default parameterless constructor as shown above, so that you never have to deal with a potentially null collection.
+
+In essence, messages should be carriers for data only. This way, discovering a bug in logic will not require redeployment of every endpoint that uses a message.
+
+
+## How do I organize messages?
+
+Because messages must be shared between multiple endpoints, you can't put the classes in the same assembly with the endpoints. They need to live in a separate class library so that they can be referenced by multiple endpoints.
```

Comment:
> Because messages must be shared between multiple endpoints, you can't put the classes in the same assembly with the endpoints

This is not correct